nemo/utils/data_utils.py

```diff
@@ -0,0 +1,289 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import pathlib
+import subprocess
+from typing import Tuple
+
+import nemo
+from nemo import constants
+from nemo.utils import logging
+
+
+def resolve_cache_dir() -> pathlib.Path:
+    """
+    Utility method to resolve a cache directory for NeMo that can be overriden by an environment variable.
+
+    Example:
+        NEMO_CACHE_DIR="~/nemo_cache_dir/" python nemo_example_script.py
+
+    Returns:
+        A Path object, resolved to the absolute path of the cache directory. If no override is provided,
+        uses an inbuilt default which adapts to nemo versions strings.
+    """
+    override_dir = os.environ.get(constants.NEMO_ENV_CACHE_DIR, "")
+    if override_dir == "":
+        path = pathlib.Path.joinpath(pathlib.Path.home(), f'.cache/torch/NeMo/NeMo_{nemo.__version__}')
+    else:
+        path = pathlib.Path(override_dir).resolve()
+    return path
+
+
+def is_store_path(path) -> bool:
+    """Check if a path is from an object store.
+    """
+    if path.startswith('ais://'):
+        return True
+    else:
+        return False
+
+
+def is_store_cache_shared() -> bool:
+    """Check if store cache is shared.
+    """
+    # Assume cache is shared by default, e.g., as in resolve_cache_dir (~/.cache)
+    cache_shared = int(os.environ.get(constants.NEMO_ENV_DATA_STORE_CACHE_SHARED, 1))
```

Comment:
Not sure how this is setup in nemo, but if that's a normal environment variable, should we add conversion to int? Since normally it would be a string if set?

Reply:
Alternative : try catch cast int, check ints, except, check str, if even that fails, give up and raise error.
Im fine with current impl.